tools/bin/ci_credentials.sh

```diff
@@ -1,26 +1,150 @@
-  [ -z "$creds" ] && error "Creds not set for $connector_name"
+  [ -z "$creds" ] && error "!!!!!Creds not set for $connector_name"
+
+  if [[ $CONNECTOR_NAME != "all" && ${connector_name} != ${CONNECTOR_NAME} ]]; then
+    return 0
+  fi
+  local key="${connector_name}#${cred_filename}"
+  [[ -z "${creds}" ]] && error "Don't find data for the connector '${key})"
+  
+  if [ -v SECRET_MAP[${key}] ]; then
+    echo "The connector '${key}' was added before"
+    return 0
+  fi
+  echo "register the secret ${key} from ${source_name}"
+  SECRET_MAP[${key}]="${creds}"
+  return 0
+}
+
+function _write_standard_creds() {
+  local connector_name=$1
+  local cred_filename=$2
+  local creds=$3
+  if jq -e . >/dev/null 2>&1 <<<${creds}; then
+      error "Failed to parse JSON for '${connector_name}' => ${cred_filename}"
+  fi
 
   if [ "$connector_name" = "base-normalization" ]; then
     local secrets_dir="airbyte-integrations/bases/${connector_name}/secrets"
   else
     local secrets_dir="airbyte-integrations/connectors/${connector_name}/secrets"
   fi
   mkdir -p "$secrets_dir"
+  echo "Saved a secret => ${secrets_dir}/${cred_filename}"
   echo "$creds" > "${secrets_dir}/${cred_filename}"
 }
 
+function save_all() {
+  for key in "${!SECRET_MAP[@]}"; do
+    local connector_name=$(echo ${key} | cut -d'#' -f1)
+    local cred_filename=$(echo ${key} | cut -d'#' -f2)
+    local creds=${SECRET_MAP[${key}]}
+    _write_standard_creds ${connector_name} ${cred_filename} "${creds}" 
+    
+  done
+  return 0
+}
+
+
+function export_github_secrets(){
+  local pairs=`echo ${SECRETS_JSON} | jq -c 'keys[] as $k | {"name": $k, "value": .[$k]} | @base64'`
+  while read row; do
+    pair=$(echo "${row}" | tr -d '"' | base64 -d)
+    local key=$(echo ${pair} | jq -r .name)  
+    local value=$(echo ${pair} | jq -r .value)
+    if [[ "$key" == *"_CREDS"* ]]; then
+      declare -gxr "${key}"="$(echo ${value})"
+    fi
+  done <<< ${pairs}
+  unset SECRETS_JSON
+}
+
+function export_gsm_secrets(){
+  local config_file=`mktemp`
+  echo "${GCP_GSM_CREDENTIALS}" > ${config_file}
+  local access_token=$(get_gcp_access_token "${config_file}" "${GSM_SCOPES}")
+  local project_id=$(parse_project_id "${config_file}")
+  rm ${config_file}
+
+  # docs: https://cloud.google.com/secret-manager/docs/filtering#api
+  local filter="name:SECRET_"
+  [[ ${CONNECTOR_NAME} != "all" ]] && filter="${filter} AND labels.connector:${CONNECTOR_NAME}"
+  local uri="https://secretmanager.googleapis.com/v1/projects/${project_id}/secrets"
+  local next_token=''
+  while true; do
+    local data=$(curl -s --get --fail "${uri}" \
+      --data-urlencode "filter=${filter}" \
+      --data-urlencode "pageToken=${next_token}" \
+      --header "authorization: Bearer ${access_token}" \
+      --header "content-type: application/json" \
+      --header "x-goog-user-project: ${project_id}")
+    [[ -z ${data} ]] && error "Can't load secrets' list"
```

Comment:
would this help in debugging failures? 
```suggestion
    [[ -z ${data} ]] && error "Can't load secret for connector ${CONNECTOR_NAME}"
```

Reply:
Done